File: mail-go/internal/routes/mail/mail_route.go
package mail

import (
	"github.com/gin-gonic/gin"
	"log"
	"mail-go/internal/services"
	"net/http"
)

type MessageRQ struct {
	Course              string `json:"course"`
	Name                string `json:"name"`
	Email               string `json:"email"`
	Company             string `json:"company"`
	Phone               string `json:"phone"`
	ParticipantsDesired string `json:"participantsDesired"`
	Date                string `json:"date"`
}

func RegisterEmailRoutes(router *gin.Engine) {
	router.POST("/send-email", sendEmail)
}

func sendEmail(c *gin.Context) {
	log.Println("sendEmail")
	var messageRQ MessageRQ
	if err := c.ShouldBindJSON(&messageRQ); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}

	err := services.SendEmail(
		messageRQ.Course,
		messageRQ.Email,
		messageRQ.Company,
		messageRQ.Phone,
		messageRQ.ParticipantsDesired,
		messageRQ.Date,
	)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to send email"})
		return
	}

	c.JSON(200, "send email completed")
}

File: mail-rust/mail/src/routes/mail/mail_route.rs
use std::io::Error;

use actix_web::{HttpResponse, post, web};
use serde::Deserialize;

use crate::services::mail_service;

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
struct MessageRQ {
    course: String,
    name: String,
    email: String,
    company: String,
    phone: String,
    participants_desired: String,
    date: String,
}


#[post("/send-email")]
pub async fn send_email(message_rq: web::Json<MessageRQ>) -> Result<HttpResponse, Error> {
    log::info!("send_email");

    match mail_service::send_email(
        &message_rq.course,
        &message_rq.email,
        &message_rq.company,
        &message_rq.phone,
        &message_rq.participants_desired,
        &message_rq.date,
    ).await {
        Ok(_) => Ok(HttpResponse::Ok().json("send email completed")),
        Err(_) => Ok(HttpResponse::InternalServerError().json("Failed to send email")),
    }
}